### ISSUE TITLE
Allow descendant combinators in CSS selectors

### ISSUE THREAD

**Issue by Encephala** [bigskysoftware/htmx#1946]
In #1720, @alexpetros mentioned we should create an issue to discuss the syntax of allowing descendant combinators in CSS, so here we go.

---

The problem is that in for instance `hx-trigger`, whitespace is significant, i.e. it separates different event modifiers in current HTMX syntax. Hence, descendant combinators, which use whitespace in CSS, yield syntax errors when getting parsed in [getTriggerSpecs](https://github.com/bigskysoftware/htmx/blob/a2830aa2df3a9f44f4ca4faac2d56de90f1a6b1f/src/htmx.js#L1237).
We can either introduce a new syntax to indicate such a descendent combinator, or we can see if it is possible to change the parsing logic to scan for descendant combinators while maintaining compatiblity with all existing event modifiers.

I myself have suggested a solution in #1913 which introduces a syntax first discussed in #569 in order to explicitly indicate the use of a descendant combinator.

A solution without any new syntax was suggested by @xhaggi in #1720 which does not involve any new syntax although CI tests haven't been run on it (in Github anyways), so we can't say for sure whether that solution is compatible with all other syntax.

**Comment by Encephala** [comment]
@xhaggi do you know if all the tests pass for your PR?

**Comment by xhaggi** [comment]
I always run the tests locally before creating a pull request, so I would say yes. But this PR needs to be rebased against dev.

**Comment by xhaggi** [comment]
FYI I did a rebase to my PR (force-pushed) and rerun all tests locally, everything is green.

**Comment by alexpetros** [comment]
Thanks for bumping this. @xhaggi's PR was functionally correct, but yes, I thought this example was way too confusing.

```html
<div hx-get="/test" hx-trigger="input from:form input"></div>
```

Not at all intuitive that one of these `input`s is a new option, and the other is a descendant combinator. I'll take a look at #1913

**Comment by Encephala** [comment]
Ass this was solved in #1913, issue closed!
